### ISSUE TITLE
Repository not found

### ISSUE THREAD

**Issue by theahura** [cpina/github-action-push-to-another-repository#75]
I was able to successfully use this action 3 days ago. Unfortunately, starting today im seeing the following error:

```
remote: Repository not found.
fatal: repository 'https://github.com/ORGNAME/REPONAME.git/' not found
```
tried resetting the personal access token I was using, but didn't seem to make any difference. Any idea what's going on?

**Comment by cpina** [comment]
Hi @theahura - I've just tried it and it worked for me (using a personal token).

Is the origin and destination repos public? Or only one? Can you link to them so I see the full action output and verify things? If you cannot link to them I'll try to send some instructions that will help (hopefully) knowing more. And the full copy-paste of the github output would help as well I think.

**Comment by theahura** [comment]
All private, unfortunately, but I ended up finding the fix: https://stackoverflow.com/a/69979203/3269537

TLDR looks like this is a known issue with github tokens in actions. Running 
`git config --unset-all http.https://github.com/.extraheader`
before running this script fixed the issue

**Comment by cpina** [comment]
Thanks @theahura for adding the solution!

Just for me to understand your solution: are you running the ` git config --unset-all http.https://github.com/.extraheader` in a step previous to the step that is using the `github-action-push-to-another-repository` action?
Thanks!

**Comment by theahura** [comment]
Yep, exactly that. It's just a normal run step

**Comment by cpina** [comment]
Hi @dlamon1 

Is your repository public? Could you paste a link of the failing GitHub Action?

Thank you

**Comment by daniel2231** [comment]
> Hi @cpina, How can we pass the value of "destination-github-username" and "destination-repository-name" as variables? also, is it possible to have the option to create a PR against the destination repo instead of actually pushing the content?

Hey @kurosh-sahraie 
If you go to the repository Settings>Secrets and variables>Actions, you will be able to set up repository specific variables.

**Comment by Cashdillo** [comment]
Hello @cpina  I am facing same problem, this is my ci/cd pipeline
cicd
=====


name: DeployToEc2
on:
   push:
      branches:
      - cicdforclientgit
jobs:
  build:
    runs-on: ubuntu-latest
    if: github.ref == 'refs/heads/cicdforclientgit'
    #container: pandoc/latex    # "ubuntu" is a more generic container
    steps:
      - uses: actions/checkout@v2
      #- name: Exclude .github and .git directory 
       # run:  apk add rsync && rsync -arv --exclude=".github" --exclude=".github"
      - name: Pushes to another repository
       # id: push_directory
        uses: cpina/github-action-push-to-another-repository@main
        env:
          API_TOKEN_GITHUB: ${{ secrets.API_TOKEN_GITHUB }}
        with:
          source-directory: .
          destination-github-username: 'Cashdillo'
          destination-repository-name: 'backend'
          user-name: arjun8010
          user-email: arjun@webmobinfo.ch
          commit-message: See ORIGIN_COMMIT from $GITHUB_REF
          target-branch: cicdtestingbranch
      - name: Test get variable exported by push-to-another-repository
        run: echo $DESTINATION_CLONED_DIRECTORY
      
can you please explain why this error occur beacuse my https://github.com/Cashdillo/backend.git repository url shown in logs is correct
this is my terminal output for your refrance
======================================
webmob@webmob-Latitude-5420:~/Downloads/new-innofund$ git clone https://github.com/Cashdillo/backend.git
Cloning into 'backend'...
Username for 'https://github.com': arjun8010
Password for 'https://arjun8010@github.com': 
remote: Enumerating objects: 904, done.
remote: Counting objects: 100% (386/386), done.
remote: Compressing objects: 100% (171/171), done.
remote: Total 904 (delta 291), reused 252 (delta 214), pack-reused 518
Receiving objects: 100% (904/904), 562.02 KiB | 2.37 MiB/s, done.
Resolving deltas: 100% (587/587), done.
webmob@webmob-Latitude-5420:~/Downloads/new-innofund$ ls -ltr
total 4
drwxrwxr-x 13 webmob webmob 4096 Mar 16 14:14 backend
webmob@webmob-Latitude-5420:~/Downloads/new-innofund$ cd backend/
webmob@webmob-Latitude-5420:~/Downloads/new-innofund/backend$ ls
app.js  constants.js  cronServices.js  emailTemplates  package.json       README.md  services  utils
auth    controllers   db.js            models          package-lock.json  server.js  socket    validators
webmob@webmob-Latitude-5420:~/Downloads/new-innofund/backend$ git fetch
Username for 'https://github.com': arjun8010
Password for 'https://arjun8010@github.com': 
webmob@webmob-Latitude-5420:~/Downloads/new-innofund/backend$ git checkout cicdtestingbranch 
Branch 'cicdtestingbranch' set up to track remote branch 'cicdtestingbranch' from 'origin'.
Switched to a new branch 'cicdtestingbranch'
webmob@webmob-Latitude-5420:~/Downloads/new-innofund/backend$ ls -ltr
total 296
drwxrwxr-x 2 webmob webmob   4096 Mar 16 14:14 services
-rw-rw-r-- 1 webmob webmob    423 Mar 16 14:14 server.js
-rw-rw-r-- 1 webmob webmob     89 Mar 16 14:14 README.md
-rw-rw-r-- 1 webmob webmob 237811 Mar 16 14:14 package-lock.json
-rw-rw-r-- 1 webmob webmob   1117 Mar 16 14:14 package.json
drwxrwxr-x 2 webmob webmob   4096 Mar 16 14:14 models
drwxrwxr-x 2 webmob webmob   4096 Mar 16 14:14 emailTemplates
-rw-rw-r-- 1 webmob webmob    566 Mar 16 14:14 db.js
-rw-rw-r-- 1 webmob webmob   2594 Mar 16 14:14 cronServices.js
drwxrwxr-x 2 webmob webmob   4096 Mar 16 14:14 controllers
-rw-rw-r-- 1 webmob webmob   3472 Mar 16 14:14 constants.js
drwxrwxr-x 2 webmob webmob   4096 Mar 16 14:14 auth
-rw-rw-r-- 1 webmob webmob   2247 Mar 16 14:14 app.js
drwxrwxr-x 2 webmob webmob   4096 Mar 16 14:14 validators
drwxrwxr-x 2 webmob webmob   4096 Mar 16 14:14 utils
drwxrwxr-x 2 webmob webmob   4096 Mar 16 14:14 socket

 
this is my error step logs 
========================
Run cpina/github-action-push-to-another-repository@main
/usr/bin/docker run --name c0442ac0b3da90d6e4538851a3c90a4b15089_290916 --label 6c0442 --workdir /github/workspace --rm -e "API_TOKEN_GITHUB" -e "INPUT_SOURCE-DIRECTORY" -e "INPUT_DESTINATION-GITHUB-USERNAME" -e "INPUT_DESTINATION-REPOSITORY-NAME" -e "INPUT_USER-NAME" -e "INPUT_USER-EMAIL" -e "INPUT_COMMIT-MESSAGE" -e "INPUT_TARGET-BRANCH" -e "INPUT_SOURCE-BEFORE-DIRECTORY" -e "INPUT_GITHUB-SERVER" -e "INPUT_DESTINATION-REPOSITORY-USERNAME" -e "INPUT_TARGET-DIRECTORY" -e "HOME" -e "GITHUB_JOB" -e "GITHUB_REF" -e "GITHUB_SHA" -e "GITHUB_REPOSITORY" -e "GITHUB_REPOSITORY_OWNER" -e "GITHUB_REPOSITORY_OWNER_ID" -e "GITHUB_RUN_ID" -e "GITHUB_RUN_NUMBER" -e "GITHUB_RETENTION_DAYS" -e "GITHUB_RUN_ATTEMPT" -e "GITHUB_REPOSITORY_ID" -e "GITHUB_ACTOR_ID" -e "GITHUB_ACTOR" -e "GITHUB_TRIGGERING_ACTOR" -e "GITHUB_WORKFLOW" -e "GITHUB_HEAD_REF" -e "GITHUB_BASE_REF" -e "GITHUB_EVENT_NAME" -e "GITHUB_SERVER_URL" -e "GITHUB_API_URL" -e "GITHUB_GRAPHQL_URL" -e "GITHUB_REF_NAME" -e "GITHUB_REF_PROTECTED" -e "GITHUB_REF_TYPE" -e "GITHUB_WORKFLOW_REF" -e "GITHUB_WORKFLOW_SHA" -e "GITHUB_WORKSPACE" -e "GITHUB_ACTION" -e "GITHUB_EVENT_PATH" -e "GITHUB_ACTION_REPOSITORY" -e "GITHUB_ACTION_REF" -e "GITHUB_PATH" -e "GITHUB_ENV" -e "GITHUB_STEP_SUMMARY" -e "GITHUB_STATE" -e "GITHUB_OUTPUT" -e "RUNNER_OS" -e "RUNNER_ARCH" -e "RUNNER_NAME" -e "RUNNER_TOOL_CACHE" -e "RUNNER_TEMP" -e "RUNNER_WORKSPACE" -e "ACTIONS_RUNTIME_URL" -e "ACTIONS_RUNTIME_TOKEN" -e "ACTIONS_CACHE_URL" -e GITHUB_ACTIONS=true -e CI=true -v "/var/run/docker.sock":"/var/run/docker.sock" -v "/home/runner/work/_temp/_github_home":"/github/home" -v "/home/runner/work/_temp/_github_workflow":"/github/workflow" -v "/home/runner/work/_temp/_runner_file_commands":"/github/file_commands" -v "/home/runner/work/innofund-backend/innofund-backend":"/github/workspace" 6c0442:ac0b3da90d6e4538851a3c90a4b15089  "" "." "Cashdillo" "backend" "github.com" "arjun@webmobinfo.ch" "arjun8010" "" "cicdtestingbranch" "See ORIGIN_COMMIT from $GITHUB_REF" ""
[+] Action start
[+] Using API_TOKEN_GITHUB
[+] Git version
git version 2.38.4
[+] Enable git lfs
Git LFS initialized.
[+] Cloning destination git repository backend
Cloning into '/tmp/tmp.liGgCn'...
total 336
drwx------   [13](https://github.com/innofund-webmob/innofund-backend/actions/runs/4435034813/jobs/7781679848#step:4:14) root     root          4096 Mar 16 08:32 .
drwxrwxrwt    1 root     root          4096 Mar 16 08:32 ..
-rw-r--r--    1 root     root            39 Mar 16 08:32 .eslintignore
-rw-r--r--    1 root     root          1123 Mar 16 08:32 .eslintrc.json
drwxr-xr-x    8 root     root          4096 Mar 16 08:32 .git
drwxr-xr-x    3 root     root          4096 Mar 16 08:32 .github
-rw-r--r--    1 root     root            25 Mar 16 08:32 .gitignore
-rw-r--r--    1 root     root            39 Mar 16 08:32 .prettierignore
-rw-r--r--    1 root     root           [14](https://github.com/innofund-webmob/innofund-backend/actions/runs/4435034813/jobs/7781679848#step:4:15)3 Mar [16](https://github.com/innofund-webmob/innofund-backend/actions/runs/4435034813/jobs/7781679848#step:4:17) 08:32 .prettierrc.js
drwxr-xr-x    2 root     root          4096 Mar 16 08:32 .vscode
-rw-r--r--    1 root     root            89 Mar 16 08:32 README.md
-rw-r--r--    1 root     root          2247 Mar 16 08:32 app.js
drwxr-xr-x    2 root     root          4096 Mar 16 08:32 auth
-rw-r--r--    1 root     root          3472 Mar 16 08:32 constants.js
drwxr-xr-x    2 root     root          4096 Mar 16 08:32 controllers
-rw-r--r--    1 root     root          2594 Mar 16 08:32 cronServices.js
-rw-r--r--    1 root     root           566 Mar 16 08:32 db.js
drwxr-xr-x    2 root     root          4096 Mar 16 08:32 emailTemplates
drwxr-xr-x    2 root     root          4096 Mar 16 08:32 models
-rw-r--r--    1 root     root        237811 Mar 16 08:32 package-lock.json
-rw-r--r--    1 root     root          11[17](https://github.com/innofund-webmob/innofund-backend/actions/runs/4435034813/jobs/7781679848#step:4:18) Mar 16 08:32 package.json
-rw-r--r--    1 root     root           423 Mar 16 08:32 server.js
drwxr-xr-x    2 root     root          4096 Mar 16 08:32 services
drwxr-xr-x    2 root     root          4096 Mar 16 08:32 socket
drwxr-xr-x    2 root     root          4096 Mar 16 08:32 utils
drwxr-xr-x    2 root     root          4096 Mar 16 08:32 validators
[+] Deleting /tmp/tmp.liGgCn//
[+] Creating (now empty) /tmp/tmp.liGgCn//
[+] Listing Current Directory Location
total 336
drwxr-xr-x   13 1001     ntp           4096 Mar 16 08:32 .
drwxr-xr-x    6 root     root          4096 Mar 16 08:32 ..
-rw-r--r--    1 1001     ntp             39 Mar 16 08:32 .eslintignore
-rw-r--r--    1 1001     ntp           1123 Mar 16 08:32 .eslintrc.json
drwxr-xr-x    8 1001     ntp           4096 Mar 16 08:32 .git
drwxr-xr-x    3 1001     ntp           4096 Mar 16 08:32 .github
-rw-r--r--    1 1001     ntp             25 Mar 16 08:32 .gitignore
-rw-r--r--    1 1001     ntp             39 Mar 16 08:32 .prettierignore
-rw-r--r--    1 1001     ntp            143 Mar 16 08:32 .prettierrc.js
drwxr-xr-x    2 1001     ntp           4096 Mar 16 08:32 .vscode
-rw-r--r--    1 1001     ntp             89 Mar 16 08:32 README.md
-rw-r--r--    1 1001     ntp           2247 Mar 16 08:32 app.js
drwxr-xr-x    2 1001     ntp           4096 Mar 16 08:32 auth
-rw-r--r--    1 1001     ntp           3472 Mar 16 08:32 constants.js
drwxr-xr-x    2 1001     ntp           4096 Mar 16 08:32 controllers
-rw-r--r--    1 1001     ntp           2594 Mar 16 08:32 cronServices.js
-rw-r--r--    1 1001     ntp            566 Mar 16 08:32 db.js
drwxr-xr-x    2 1001     ntp           4096 Mar 16 08:32 emailTemplates
drwxr-xr-x    2 1001     ntp           4096 Mar 16 08:32 models
-rw-r--r--    1 1001     ntp         237811 Mar 16 08:32 package-lock.json
-rw-r--r--    1 1001     ntp           1117 Mar 16 08:32 package.json
-rw-r--r--    1 1001     ntp            423 Mar 16 08:32 server.js
drwxr-xr-x    2 1001     ntp           4096 Mar 16 08:32 services
drwxr-xr-x    2 1001     ntp           4096 Mar 16 08:32 socket
drwxr-xr-x    2 1001     ntp           4096 Mar 16 08:32 utils
drwxr-xr-x    2 1001     ntp           4096 Mar 16 08:32 validators
[+] Listing root Location
total 84
drwxr-xr-x    1 root     root          4096 Mar 16 08:32 .
drwxr-xr-x    1 root     root          4096 Mar 16 08:32 ..
-rwxr-xr-x    1 root     root             0 Mar 16 08:32 .dockerenv
drwxr-xr-x    2 root     root          4096 Feb 10 16:45 bin
drwxr-xr-x    5 root     root           340 Mar 16 08:32 dev
-rwxr-xr-x    1 root     root          5355 Feb 25 [21](https://github.com/innofund-webmob/innofund-backend/actions/runs/4435034813/jobs/7781679848#step:4:22):58 entrypoint.sh
drwxr-xr-x    1 root     root          4096 Mar 16 08:32 etc
drwxr-xr-x    6 root     root          4096 Mar 16 08:32 github
drwxr-xr-x    2 root     root          4096 Feb 10 16:45 home
drwxr-xr-x    1 root     root          4096 Feb 10 16:45 lib
drwxr-xr-x    5 root     root          4096 Feb 10 16:45 media
drwxr-xr-x    2 root     root          4096 Feb 10 16:45 mnt
drwxr-xr-x    2 root     root          4096 Feb 10 16:45 opt
dr-xr-xr-x  210 root     root             0 Mar 16 08:32 proc
drwx------    2 root     root          4096 Feb 10 16:45 root
drwxr-xr-x    1 root     root          4096 Mar 16 08:32 run
drwxr-xr-x    2 root     root          4096 Feb 10 16:45 sbin
drwxr-xr-x    2 root     root          4096 Feb 10 16:45 srv
dr-xr-xr-x   12 root     root             0 Mar 16 08:32 sys
drwxrwxrwt    1 root     root          4096 Mar 16 08:32 tmp
drwxr-xr-x    1 root     root          4096 Mar 16 08:32 usr
drwxr-xr-x    1 root     root          4096 Mar 16 08:32 var
[+] List contents of .
README.md
app.js
auth
constants.js
controllers
cronServices.js
db.js
emailTemplates
models
package-lock.json
package.json
server.js
services
socket
utils
validators
[+] Checking if local . exist
[+] Copying contents of source repository folder . to folder  in git repo backend
[+] Files that will be pushed
total 336
drwxr-xr-x   13 1001     ntp           4096 Mar 16 08:32 .
drwxrwxrwt    1 root     root          4096 Mar 16 08:32 ..
-rw-r--r--    1 1001     ntp             39 Mar 16 08:32 .eslintignore
-rw-r--r--    1 1001     ntp           1123 Mar 16 08:32 .eslintrc.json
drwxr-xr-x    8 1001     ntp           4096 Mar 16 08:32 .git
drwxr-xr-x    3 1001     ntp           4096 Mar 16 08:32 .github
-rw-r--r--    1 1001     ntp             25 Mar 16 08:32 .gitignore
-rw-r--r--    1 1001     ntp             39 Mar 16 08:32 .prettierignore
-rw-r--r--    1 1001     ntp            143 Mar 16 08:32 .prettierrc.js
drwxr-xr-x    2 1001     ntp           4096 Mar 16 08:32 .vscode
-rw-r--r--    1 1001     ntp             89 Mar 16 08:32 README.md
-rw-r--r--    1 1001     ntp           [22](https://github.com/innofund-webmob/innofund-backend/actions/runs/4435034813/jobs/7781679848#step:4:23)47 Mar 16 08:32 app.js
drwxr-xr-x    2 1001     ntp           4096 Mar 16 08:32 auth
-rw-r--r--    1 1001     ntp           3472 Mar 16 08:32 constants.js
drwxr-xr-x    2 1001     ntp           4096 Mar 16 08:32 controllers
-rw-r--r--    1 1001     ntp           2594 Mar 16 08:32 cronServices.js
-rw-r--r--    1 1001     ntp            566 Mar 16 08:32 db.js
drwxr-xr-x    2 1001     ntp           4096 Mar 16 08:32 emailTemplates
drwxr-xr-x    2 1001     ntp           4096 Mar 16 08:32 models
-rw-r--r--    1 1001     ntp         [23](https://github.com/innofund-webmob/innofund-backend/actions/runs/4435034813/jobs/7781679848#step:4:24)7811 Mar 16 08:[32](https://github.com/innofund-webmob/innofund-backend/actions/runs/4435034813/jobs/7781679848#step:4:33) package-lock.json
-rw-r--r--    1 1001     ntp           1117 Mar 16 08:32 package.json
-rw-r--r--    1 1001     ntp            423 Mar 16 08:32 server.js
drwxr-xr-x    2 1001     ntp           [40](https://github.com/innofund-webmob/innofund-backend/actions/runs/4435034813/jobs/7781679848#step:4:41)96 Mar 16 08:32 services
drwxr-xr-x    2 1001     ntp           4096 Mar 16 08:32 socket
drwxr-xr-x    2 1001     ntp           4096 Mar 16 08:32 utils
drwxr-xr-x    2 1001     ntp           4096 Mar 16 08:32 validators
[+] Set directory is safe (/tmp/tmp.liGgCn)
[+] Adding git commit
[+] git status:
On branch cicdforclientgit
Your branch is up to date with 'origin/cicdforclientgit'.

nothing to commit, working tree clean
[+] git diff-index:
[+] Pushing git commit
remote: Repository not found.
fatal: repository 'https://github.com/Cashdillo/backend.git/' not found

**Comment by cpina** [comment]
https://github.com/Cashdillo/backend.git/ is private, so the git action (in your case) is trying to use the API_TOKEN_GITHUB. My only idea is that the API_TOKEN_GITHUB is wrong / expired. In the "troubleshotting" page there is a section to help to diagnose the problem (validate teh API_TOKEN_GITHUB):
https://cpina.github.io/push-to-another-repository-docs/troubleshooting.html#test-the-personal-access-token

That might help?

**Comment by Cashdillo** [comment]
@cpina but i already sent you my terminal output in which i am using same token & this token have all the access like admin

**Comment by Cashdillo** [comment]
@cpina i had use this command for troubleshoot the problem 
git clone https://YOUR_USERNAME:YOUR_ACCESS_TOKEN@github.com/SOMETHING/REPO
So in the output repo is cloning successfully & push & pull both command are working properly, so i think is there any other issue

**Comment by cpina** [comment]
Make sure that your local git config does not have something like:
```
[url "git@github.com:"]
	insteadOf = https://github.com/
```
This confused me once: I was trying to use the https:// endpoint and it was being re-written to git://

I will make a branch to add debugging information on the "git push". If you want to try yourself before you can fork and add in the git push line:

`GIT_CURL_VERBOSE=1 GIT_TRACE=1 git push -v # rest of the line

To see if you can see any useful information.

I've just tried with one of my repositories and I didn't have any problem on my side (using the access token).

**Comment by cpina** [comment]
Try using the github action in the @debug branch. Such as:
```
uses: cpina/github-action-push-to-another-repository@debug
```
Copy-paste the output.

I haven't used the debug options before, make sure that no personal information or credentials are leaked.

This should help finding the reason that the destination repository is not found... finger crossed!
